### ISSUE TITLE
target_mask.sum() == 0 causing NaN loss during training (debug & potential fix)

### ISSUE THREAD

**Issue by RongLirr** [sign-language-processing/fluent-pose-synthesis#7]
While training my model with a couple of real pose samples, I encountered a NaN issue caused by the following situation:

target_mask.sum() == 0
→ model receives no valid supervision
→ loss = NaN

This was traced back to my `load_data.py`. In my current dataset implementation (`SignLanguagePoseDataset`), I extracted the `target_mask` from `fluent_pose.body.mask`, which is built internally using the confidence == 0 logic in `NumPyPoseBody`.

This mask is at joint level and does not guarantee that a frame contains any valid joints. I think this is why some randomly sampled `fluent_clip` windows consisted entirely of all-zero frames. After padding with `zero_pad_collator`, `target_mask.sum()`became zero. Here's what the collated batch looked like at runtime:

```
Dataset initialized with 128 samples. Split: train
Target clip shape: torch.Size([32, 50, 1, 178, 3])
Input clip shape: torch.Size([32, 282, 1, 178, 3])
Input mask shape: torch.Size([32, 282, 1, 178, 3])
Target mask shape: torch.Size([32, 50, 1, 178, 3])
Sample 0 target_mask sum: 0
Sample 1 target_mask sum: 0
Sample 2 target_mask sum: 0
Sample 3 target_mask sum: 1
Sample 4 target_mask sum: 0
Sample 5 target_mask sum: 0
Sample 6 target_mask sum: 0
...
```

Potential Fix:
Instead of using `pose.body.mask`, compute per-frame mask directly from data
`target_mask = np.any(fluent_clip != 0, axis=(1, 2, 3))  # shape: [T]`

In this way, each frame in the fluent_clip is marked valid only if at least one joint is non-zero. Then the result became:

```
Target clip shape: torch.Size([32, 80, 1, 178, 3])
Input clip shape: torch.Size([32, 269, 1, 178, 3])
Input mask shape: torch.Size([32, 269, 1, 178, 3])
Target mask shape: torch.Size([32, 80])
Sample 0 target_mask sum: 51
Sample 1 target_mask sum: 34
Sample 2 target_mask sum: 19
Sample 3 target_mask sum: 28
Sample 4 target_mask sum: 40
Sample 5 target_mask sum: 80
Sample 6 target_mask sum: 20
Sample 7 target_mask sum: 42
Sample 8 target_mask sum: 80
Sample 9 target_mask sum: 36
...
```

I’ll experiment with the training pipeline later to verify whether everything works.
